lib/Parser/Parser.cpp

```diff
@@ -3332,6 +3332,16 @@ class CustomOpAsmParser : public OpAsmParser {
   ParseResult parseOperandList(SmallVectorImpl<OperandType> &result,
                                int requiredOperandCount = -1,
                                Delimiter delimiter = Delimiter::None) override {
+    return parseOperandOrRegionArgList(result, /*isOperandList=*/true,
+                                       requiredOperandCount, delimiter);
+  }
+
+  /// Parse zero or more SSA comma-separated operand references with a specified
+  /// surrounding delimiter, and an optional required operand count.
+  ParseResult
+  parseOperandOrRegionArgList(SmallVectorImpl<OperandType> &result,
+                              bool isOperandList, int requiredOperandCount = -1,
```

Comment:
I'm not a fan of isOperandList: its enabling code reuse but at the cost of the modelling IMHO. Could you try a different refactoring that enables the reuse (most of the reuse) but avoids the flag & if below?

Reply:
I had the same thoughts. But I couldn't find another refactoring that was clean and allowed all / most of the reuse. If we think about it, the parsing is really the same for both except for how each element is treated (for one, the SSA use has to be resolved, and for the other, the absence of def's to the same name has to be checked); so this made sense.